kernel/src/actions/crc.rs

```diff
@@ -271,7 +308,120 @@ mod tests {
             ]),
         };
 
-        assert_eq!(visitor.protocol, expected_protocol);
-        assert_eq!(visitor.metadata, expected_metadata);
+        assert_eq!(visitor.protocol, Some(expected_protocol));
+        assert_eq!(visitor.metadata, Some(expected_metadata));
+    }
+
+    #[tokio::test]
+    async fn pm_try_new() {
+        let crc_json = serde_json::json!({
+            "tableSizeBytes": 100,
+            "numFiles": 10,
+            "numMetadata": 1,
+            "numProtocol": 1,
+            "metadata": {
+                "id": "testId",
+                "format": {
+                    "provider": "parquet",
+                    "options": {}
+                },
+                "schemaString": r#"{"type":"struct","fields":[{"name":"value","type":"integer","nullable":true,"metadata":{}}]}"#,
+                "partitionColumns": [],
+                "configuration": {
+                    "delta.columnMapping.mode": "none"
+                },
+                "createdTime": 1677811175
+            },
+            "protocol": {
+                "minReaderVersion": 3,
+                "minWriterVersion": 7,
+                "readerFeatures": ["columnMapping"],
+                "writerFeatures": ["columnMapping"]
+            }
+        });
+
+        let json = crc_json.to_string().into_bytes();
+        // let (engine, store) = DefaultEngine::new_memory();
```

Comment:
anyone know why I can't seem to get the extension trait working here?

Reply:
We know now and can remove the commented out line (sadly)